server/src/main/java/org/opensearch/common/util/ValidationUtils.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+/**
+ * Utils to do validations
+ *
+ * @opensearch.internal
+ */
+public final class ValidationUtils {
```

Comment:
Please remove this class, it is not very useful

Reply:
this exactly acts as https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkArgument(boolean) so I think it is useful

Reply:
It gives no indication of what is wrong and why exception was raised

Reply:
yep I agree. I've added message as param to the method so caller can pass a message

Reply:
I would argue, we don't need such classes, you use it in just 2 places but introduce a whole class for that, what is the purpose?

Reply:
same reason why Guava Preconditions exists, it a utility class for validations. Yes it is used in our code base in 2 places right now but I believe this pattern exist everywhere and having such utility class and start using it will enhance code cohesion

Reply:
I will try to be as constructive as possible: we are developing against public codebase, there is no reason for accidental classes like this to exists without practicality (btw, the codebase doesn't use `guava`, for a reason). You want to use method indirection in instead of simple `if` - move this method to the consumer class.

Reply:
+1 💯 agree w/ @reta. Move `checkArgument` as a private static method in SegmentedCache.java and remove this class. It's superfluous.